pyuvdata/uvcal/initializers.py

```diff
@@ -0,0 +1,529 @@
+    XORIENTMAP,
+    Locations,
+    get_antenna_params,
+    get_freq_params,
+    get_spw_params,
+    get_time_params,
+)
+
+
+def new_uvcal(
+    *,
+    time_array: np.ndarray,
+    antenna_positions: np.ndarray | dict[str | int, np.ndarray],
+    telescope_location: Locations,
+    telescope_name: str,
+    cal_style: Literal["sky", "redundant"],
+    gain_convention: Literal["divide", "multiply"],
+    jones_array: np.ndarray | str,
+    x_orientation: Literal["east", "north", "e", "n", "ew", "ns"],
+    freq_array: np.ndarray | None = None,
+    freq_range: np.ndarray | None = None,
+    cal_type: Literal["delay", "gain"] | None = None,
+    integration_time: float | np.ndarray | None = None,
+    channel_width: float | np.ndarray | None = None,
+    antenna_names: list[str] | None = None,
+    antenna_numbers: list[int] | None = None,
+    antname_format: str = "{0:03d}",
+    ant_array: np.ndarray | None = None,
+    wide_band: bool = False,
+    flex_spw_id_array: np.ndarray | None = None,
+    ref_antenna_name: str | None = None,
+    sky_catalog: str | None = None,
+    sky_field: str | None = None,
+    empty: bool = False,
+    data: dict[str, np.ndarray] | None = None,
+    history: str = "",
+    **kwargs,
+):
+    """Create a new UVCal object with default parameters.
+
+    Parameters
+    ----------
+    freq_array : ndarray of float
+        Array of frequencies in Hz.
+    time_array : ndarray of float
+        Array of times in Julian Date.
+    antenna_positions : ndarray of float or dict of ndarray of float
+        Array of antenna positions in ECEF coordinates in meters.
+        If a dict, keys can either be antenna numbers or antenna names, and values are
+        position arrays. Keys are interpreted as antenna numbers if they are integers,
+        otherwise they are interpreted as antenna names if strings. You cannot
+        provide a mix of different types of keys.
+    telescope_location : ndarray of float or str
+        Telescope location as an astropy EarthLocation object or MoonLocation object.
+    telescope_name : str
+        Telescope name.
+    cal_style : str
+        Calibration style. Options are 'sky' or 'redundant'.
+    gain_convention : str
+        Gain convention. Options are 'divide' or 'multiply'.
+    x_orientation : str
+        Orientation of the x-axis. Options are 'east', 'north', 'e', 'n', 'ew', 'ns'.
+    jones_array : ndarray of int or str
+        Array of Jones polarization integers. If a string, options are 'linear' or
+        'circular' (which will be converted to the appropriate Jones integers as
+        length-4 arrays).
+    delay_array : ndarray of float, optional
+        Array of delays in seconds. If cal_type is not provided, it will be set to
+        'delay' if delay_array is provided.
+    cal_type : str, optional
+        Calibration type. Options are 'delay', 'gain', or 'unknown'. Not required
+        if delay_array is provided (then it will be set to 'delay').
```

Comment:
we've deprecated the `unknown` cal_type, so let's not let it be set to unknown here.

Reply:
Ah, I removed that option from the code, but not the docs. Good spotting.